src/chat/chatAgentRequestHandler.ts

```diff
@@ -0,0 +1,633 @@
+            const editorToUse =
+                activeEditor || (await findEditorFromReferences(request.references));
+
+            // Use the preferred editor's URI if available
+            connectionUri = editorToUse?.document.uri.toString() ?? connectionUri;
+
+            for (const reference of request.references) {
+                const value = reference.value;
+                if (value instanceof vscode.Location) {
+                    // Could be a document / selection in the current editor
+                    if (
+                        editorToUse &&
+                        value.uri.toString() === editorToUse.document.uri.toString()
+                    ) {
+                        referenceTexts.push(
+                            `${reference.modelDescription ?? "ChatResponseReference"}: ${editorToUse.document.getText(value.range)}`,
+                        );
+                    } else {
+                        const doc = await vscode.workspace.openTextDocument(value.uri);
+                        referenceTexts.push(
+                            `${reference.modelDescription ?? "ChatResponseReference"}: ${doc.getText(value.range)}`,
+                        );
+                    }
+                } else if (value instanceof vscode.Uri) {
+                    // Could be a file/document
+                    const doc = await vscode.workspace.openTextDocument(value);
+                    referenceTexts.push(
+                        `${reference.modelDescription ?? "ChatResponseReference"}: ${doc.getText()}`,
+                    );
+                } else if (typeof reference.value === "string") {
+                    // Could be a string
+                    referenceTexts.push(
+                        `${reference.modelDescription ?? "ChatResponseReference"}: ${reference.value}`,
+                    );
+                }
+            }
+        }
+
+        const prompt = request.prompt.trim();
+        const [model] = await vscode.lm.selectChatModels(MODEL_SELECTOR);
+
+        try {
+            if (!model) {
+                stream.markdown("No model found.");
+                return { metadata: { command: "", correlationId: correlationId } };
+            }
+
+            // Tool lookup
+            const copilotDebugLogging = vscodeWrapper
+                .getConfiguration()
+                .get(Constants.copilotDebugLogging, false);
+            if (copilotDebugLogging) {
+                stream.progress(
+                    `Using ${model.name} (${context.languageModelAccessInformation.canSendRequest(model)})...`,
+                );
+            }
+
+            if (!connectionUri) {
+                await sendToDefaultLanguageModel(prompt, model, stream, token);
+                return { metadata: { command: "", correlationId: correlationId } };
+            }
+
+            const success = await copilotService.startConversation(
```

Comment:
do details on the nature of this failure get captured elsewhere?

Reply:
it looks like there is a telemetry event below, but give the failure code is a boolean, what additional context should be captured?  There are error logs in STS if that's what you are asking about.

Reply:
Yeah, STS logs are good enough.  MIght've expected to see that bubble higher up as well but not worth changing the contract.